components/core/src/fs.rs

```diff
@@ -41,10 +36,24 @@ pub const CACHE_SRC_PATH: &'static str = "hab/cache/src";
 pub const CACHE_SSL_PATH: &'static str = "hab/cache/ssl";
 /// The root path containing all locally installed packages
 pub const PKG_PATH: &'static str = "hab/pkgs";
+/// The environment variable pointing to the filesystem root. This exists for internal
+/// Habitat team usage and is not intended to be used by Habitat consumers.
+/// Using this variable could lead to broken supervisor services and it should
+/// be used with extreme caution.
+pub const FS_ROOT_ENVVAR: &'static str = "FS_ROOT";
 /// The root path containing all runtime service directories and files
 const SVC_PATH: &'static str = "hab/svc";
 
 lazy_static! {
+    /// The default filesystem root path
+    pub static ref FS_ROOT_PATH: String = {
+        if cfg!(target_os = "windows") && henv::var(FS_ROOT_ENVVAR).is_ok() {
```

Comment:
That was my initial approach @fnichol but the compiler was seeing them as duplicate declarations and I am guessing because `lazy_static!` is a macro and something funky was happening by nesting those macros. Here is the error:
```
error[E0428]: a value named `FS_ROOT_PATH` has already been defined in this module
   --> src\fs.rs:47:1
    |
47  | lazy_static! {
    | ^
    | |
    | previous definition of `FS_ROOT_PATH` here
    | already defined
    |
    = note: this error originates in a macro outside of the current crate

error[E0119]: conflicting implementations of trait `std::ops::Deref` for type `fs::FS_ROOT_PATH`:
   --> src\fs.rs:47:1
    |
47  | lazy_static! {
    | ^
    | |
    | first implementation here
    | conflicting implementation for `fs::FS_ROOT_PATH`
    |
    = note: this error originates in a macro outside of the current crate

error: aborting due to previous error

error: Could not compile `habitat_core`.
```
So I went with the above implementation instead.